### ISSUE TITLE
Pathways, Water, And other stuff able to be seen through terrain on Raspberry Pi 3 in Zelda Ocarina of time.

### ISSUE THREAD

**Issue by Oxreim** [gonetz/GLideN64#1688]
Example of the graphical glitches, Any help would be appreciated. Thank you!

https://imgur.com/a/Ue0hY

![s8rdpih](https://user-images.githubusercontent.com/24662815/34076602-4ec34608-e2ba-11e7-8f8e-f4a81ff22635.jpg)
![wa5hlqr](https://user-images.githubusercontent.com/24662815/34076603-4ece8f90-e2ba-11e7-857b-62e89917260d.jpg)
![wbvnnmi](https://user-images.githubusercontent.com/24662815/34076604-4ed9769e-e2ba-11e7-84f9-1e37a443cd02.jpg)

**Comment by gonetz** [comment]
Try to enable 'legacy blending' option

**Comment by Oxreim** [comment]
Ok after about 10 minutes of searching on google i give up and am just going to ask. Where do i enable legacy blending? All i found was discussion where the option was added and other discussions about it with no one saying how to enable it or giving any clue.

**Comment by gonetz** [comment]
You need to find, where mupen64plus keeps its config file. It is usually in home folder on Linux, I don't remember exactly. Something like ~/.config/mupen64plus/mupen64plus.cfg
Open it in editor, find [Video-GLideN64] section, find EnableLegacyBlending option in that section and set it to True:
`EnableLegacyBlending = True`

**Comment by Oxreim** [comment]
Oh, Glanced through that file and apparently missed it somehow in that file. Opps.

Anyway, Doesn't look like it helped with the issue, Still able to see the stuff through the walls and all with legacy blending on.

**Comment by Oxreim** [comment]
Quite simply how i am working around the issue with the latest builds is to use a slightly older build that was made before the issues detailed in #1627

**Comment by dankcushions** [comment]
If you're running via retropie (as I assume you are with the 'updating via binary' comments), you have to be mindful that you are not running 'standard' GLideN64. for example, various settings (like legacy blending) are controlled by the launcher script and it won't matter what you set them to in the cfg. eg https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/emulators/mupen64plus/mupen64plus.sh#L263

further, patches may or may not be applied to the code when compiling via source (eg that rpi-yuv.diff patch), or specific versions (non master) maybe compiled.

in short, these sorts of support issues should be sent to the retropie forums, not here.

**Comment by Metalwario64** [comment]
For some reason my Pi 3 seemed to have had trouble updating either the update script or Mupen64plus for whatever reason, since I tried both multiple times earlier today and neither worked, but now suddenly I was able to update and the Zeldas are playable again. 

The legacy blending issue still remains, and I posted a thread for it in the Retropie forums, if you were including that when you said "these sorts of support issues".

**Comment by dankcushions** [comment]
Some history on this one. At one point, we were able to avoid the zelda depth issues by showing the FPS counter. My working theory is that perhaps this line is unintentionally fixing these depth issues on videcore: https://github.com/gonetz/GLideN64/blob/9409abd671c22246291153cf919018d1096f91fc/src/GraphicsDrawer.cpp#L1457

i was just having a look further into that, but i noticed the 'fix' doesn't work any more - we have zelda depth issues even with the fps counter showing. i bisected it, and it stopped working here: https://github.com/gonetz/GLideN64/commit/84cf648089df60d7e7c04efa59cb7ae405fe7a3f

so @gonetz, can you think why that commit would affect things? i would love to get some clues as this strange behaviour. perhaps we can finally fix zelda depth for RPI videocore without the fps counter 'fix'...

**Comment by gonetz** [comment]
I'll try to find some time and investigate it, but I can't promise fast results.

**Comment by dankcushions** [comment]
Thank you @gonetz! If you need any testing on RPI3 I can do - the issue as I've been experiencing is during the title intro of Zelda: OOT. you can see tree models through the ground during the second shot, as it it pans across the Hyrule Field.

**Comment by dankcushions** [comment]
I notice this issue does not happen in the MESA RPI3 driver, which is also GLES 2 and using the same settings, so this issue is either an issue in the closed-source boardcom driver, or a side effect from the various ``#ifdef VC` bits in the code.
